ledger/ledger-api-auth/src/main/scala/com/digitalasset/ledger/api/auth/Authorizer.scala

```diff
@@ -309,8 +307,8 @@ final class Authorizer(
       authorized: (ClaimSet.Claims, Req) => Either[StatusRuntimeException, Req]
   ): Req => Future[Res] = request =>
     authenticatedClaimsFromContext() match {
-      case Failure(ex) => Future.failed(ex)
-      case Success(claims) =>
+      case Left(e: StatusRuntimeException) => Future.failed(e)
```

Comment:
Because we're using `Either[StatusRuntimeException, A]` rather than `Try[A]` we can be precisely what exception we're expecting. If the exception type would be widen for whatever reason then the compiler will warn about non-exhaustive pattern matching